x-pack/legacy/plugins/fleet/server/repositories/policies/default.ts

```diff
@@ -14,9 +15,17 @@ export class PoliciesRepository implements PoliciesRepositoryType {
    *
    * @param id
    */
-  async getFullPolicy(id: string) {
+  async getPolicyOutputByIDs(user: FrameworkUser, ids: string[]) {
     if (this.plugin) {
-      return await this.plugin.getFull(id);
+      return await this.plugin.getPolicyOutputByIDs(user, ids);
```

Comment:
I believe all the instances of `return await ...` can be converted to `return ...`.  

The exception is if the `return await` is inside a `try/catch` of another `async function`. I saw something like that in this PR at https://github.com/elastic/kibana/pull/50817/files#diff-9a99afd34ab284b44d4fe4c52301c84bR143

```typescript
    try {
      const savedObject = await this.getClient(user).get(type, id, options);

      return savedObject;
    } catch (err) {
      if (err.isBoom && err.output.statusCode === 404) {
        return null;
      }

      throw err;
    }
```

Perhaps there are others, but the default action, IMO, should be to simply `return promiseReturningAction`.

There's more on this at https://eslint.org/docs/rules/no-return-await and in [a recent EPM PR](https://github.com/elastic/kibana/pull/50329)

Reply:
🤔 It is an interesting pattern to do it this way, but I do think I tend to agree with the reason not to "to denote a value that is a thenable". Though it does create a scenario where the code runs slower... so perhaps it's not worth it... Also without the check for unhandled promises enabled, would that not lead to having potential errors that are harder to catch?
all in all I am hesitant to try to follow a pattern like this when linting is not enabled for it... (That said, we have many linting rules that exist for style reasons, and this has true perf benefits, so I am more or less pro-rule)

Reply:
having `return await` allow better stack trace and I am pretty sure that it come with almost zero cost.

Reply:
The two are identical behaviorally for any consumers (minus the noted exception about doing `return await` in an `try/catch)`.

```typescript
async function x() {
 const a = await someOtherFunction()
 return await aFunction(a)
}
```

Looking at the last line, `x`:
1. waits for the promise from `aFunction` to resolve
1. unwraps the resolved value
1. immediately hands it to `return`
1. which _rewraps_ the value back into a Promise

Versus returning the Promise directly:
```typescript
async function y() {
 const a = await someOtherFunction()
 return aFunction(a)
}
```

Which does not wait, nor unwrap just to immediately rewrap the untouched value.

There is no difference from the consumer side. Results will succeed or throw the same for both.

```typescript
async function useAsync() {
  const value1 = await x();
  const value2 = await y();
}
```

It really is just extra typing for the dev and extra time work for the JS engine. `await` is to make asynchronous code look synchronous. Particularly to make it easier to deal with the result of asynchronous operations. But we're not using the result of the async operations in this case, we're just passing it along. We can continue to pass it along without the `await`.

I'm not saying this must be changed. It's not on the critical path for this PR or our features, so merge what's here whenever you're comfortable. I'm just pointing out that it's unnecessary. I don't see a benefit to it keeping it or risk from removing it.

Reply:
@jfsiii all good points to know, so thank you for the information. I would add though that from a personal standpoint I feel it makes it easier to read/understand the code when awaiting it.

Regardless... I wanted to see what the performance implications were...

On a flat out check it was a negligible amount of ~1ms per awaited function.

I am sure there are corner cases especially in compiled code here where removing the await could prevent bugs. However given the perf implications I would say it is currently a premature optimization that need not be considered at this time? 
Especially considering there is a linting rule to find this that is currently disabled? Again, I would not debate having a linting rule for consistency sake at the very least.